modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -3130,6 +3143,50 @@ public DistributedTransactionConfiguration getDistributedTransactionConfiguratio
         return distributedTransactionConfiguration;
     }
 
+    /** @param transform Transaction message transformer. */
+    public void txMessageTransformer(BiFunction<GridCacheMessage, IgniteInternalTx, GridCacheMessage> transform) {
+        txMsgTransform = transform;
+    }
+
+    /** @param callback Callback invoked on transaction commit. */
+    public void onCommitCallback(Consumer<IgniteInternalTx> callback) {
+        onCommitCallback = callback;
+    }
+
+    /**
+     * Sends transaction message after transforming it.
+     *
+     * @param nodeId Node ID to send message.
+     * @param msg Original message to transform.
+     * @param tx Transaction.
+     * @param plc IO policy.
+     */
+    public void sendTransactionMessage(UUID nodeId, GridCacheMessage msg, IgniteInternalTx tx, byte plc) throws IgniteCheckedException {
+        BiFunction<GridCacheMessage, IgniteInternalTx, GridCacheMessage> transform = txMsgTransform;
+
+        if (transform != null)
+            msg = transform.apply(msg, tx);
+
+        cctx.io().send(nodeId, msg, plc);
+    }
+
+    /**
+     * Sends transaction message after transforming it.
+     *
+     * @param n Node to send message.
+     * @param msg Original message to transform.
+     * @param tx Transaction.
+     * @param plc IO policy.
+     */
+    public void sendTransactionMessage(ClusterNode n, GridCacheMessage msg, IgniteInternalTx tx, byte plc) throws IgniteCheckedException {
+        BiFunction<GridCacheMessage, IgniteInternalTx, GridCacheMessage> transform = txMsgTransform;
```

Comment:
Can we call `sendTransactionMessage(n.id(), msg, tx, plc)` instead of having double implementation?